lib/pages/home_page.dart

```diff
@@ -13,54 +14,131 @@ class HomePage extends StatefulWidget {
 }
 
 class _HomePageState extends State<HomePage> {
-  LocationPermission? gpsPermission;
-  Position? position;
+  LocationPermission? gps = LocationPermission.denied;
+  Position? pos;
 
-  @override
-  void initState() {
-    foo();
+  Future<void> foo() async {
+    gps = await Geolocation.instance.gpsPermission;
+    pos = await Geolocation.instance.position;
   }
 
-  Future<void> foo() async {
-    LocationPermission gps = await Geolocation().turnOnServices();
-    Position pos = await Geolocator.getCurrentPosition(
-        desiredAccuracy: LocationAccuracy.high);
-    setState(() {
-      gpsPermission = gps;
-      position = pos;
-    });
+  Future<void> refreshBtn() async {
+    setState((){});
   }
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: const Text('Home'),
-      ),
-      body: SizedBox(
-        height: double.infinity,
-        width: double.infinity,
-        child: Scaffold(
-          body: FlutterMap(
-            options: MapOptions(
-                center: LatLng(position.latitude, position.longitude),
-                zoom: 13.0,
-                minZoom: 2,
-                maxZoom: 18.3,
-                interactiveFlags: InteractiveFlag.all & ~InteractiveFlag.rotate,
-                maxBounds:
-                    LatLngBounds(LatLng(-90, -180.0), LatLng(90.0, 180.0))),
-            children: [
-              TileLayer(
-                urlTemplate:
-                    "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
-                subdomains: const ['a', 'b', 'c'],
+    return FutureBuilder(
```

Comment:
U struktury této stránky sis malinko zavařil tím, že kdykoliv proběhne build, tak se zavolá ten future a proběhne ten loading. Takže pokud chceš ten state management řešit takto, doporučil bych ti spíš ten permission pohlídat ještě v nějaké mezistránce mezi loginem a home stránkou. Můžem zkonzultovat na meetu ;)